torch_brain/utils/stitcher.py

```diff
@@ -71,7 +74,159 @@ def stitch(timestamps: torch.Tensor, values: torch.Tensor) -> torch.Tensor:
     return averages
 
 
-class StitchEvaluator(L.Callback):
+class MultiSessionDecodingStitchEvaluator(L.Callback):
```

Comment:
It is specific to multi-session situations because it requires the tokenizer to add `"session_id"` field to the batch-data. In single session situations users shouldn't be expected to add this field.

I believe having the `Stitcher` class (independent of Lightning, the idea we discussed on slack) would end up making this situation more unified. This would take some thinking and experimenting; which I can probably get to next week, but this week I'm tight on time. Until then, I feel this is a good solution.

Reply:
I agree with the second point. 

Re: `"session_id"`, I also agree. My solution is that instead of rewritting the whole class, we just add an argument to the existing one that would be "single_session=True". And it will not expect to get "session_id" in that case. everything else will be exactly the same.

Reply:
Removed "MultiSession" from the name